### ISSUE TITLE
missing node in org-roam-node-find after capture

### ISSUE THREAD

**Issue by dirodriguezm** [org-roam/org-roam#1962]
#### Steps to Reproduce
1. Run `org-roam-node-find` and create node with template "project version"
2. Run `org-roam-node-find` again
    - node appears in popup buffer
3. Run `org-roam-capture` with "project task" template
    - text gets captured fine
4. Run `org-roam-node-find` again
    - created node is missing

These are my templates
```
(setq org-roam-capture-templates
        '(("d" "default" plain "%?"
           :if-new (file+head "%<%Y%m%d%H%M%S>-${slug}.org"
                              "#+title: ${title}\n")
           :unnarrowed t)
        ("p" "project")
        ("pp" "project" plain
         (file "~/Dropbox/org_templates/project_template.org")
         :if-new (file "${slug}.org")
         :unnarrowed t)
        ("pv" "version" plain
         (file "~/Dropbox/org_templates/project_version_template.org")
         :if-new (file "${slug}.org")
         :unnarrowed t)
        ("pt" "project task" entry "** TODO %?"
         :if-new (file+olp "${slug}.org" ("Tasks")))))
```

#### Backtrace
No error

#### Expected Results

created node shows in `org-roam-node-find`
#### Actual Results
created node missing from list after capture

### Environment
```
- Emacs: GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.27, cairo version 1.17.4)
 of 2021-03-26
- Framework: Doom
- Org: Org mode version 9.6 (9.6-??-2e99997 @ /home/diego/.emacs.d/.local/straight/build-27.2/org/)
- Org-roam: 3e47f19
```

- Org-roam commit: https://github.com/jethrokuan/org-roam/commit/3e47f198c7b6c3254944d98357e41840e5e1b102

**Comment by nobiot** [comment]
As we do not have the project templates in your dropbox, I did the following.
I don't seem to be able to reproduce your issue (?)

I created a project version, and two tasks.
I ended up creating three files -- I have a feeling you would like to create the tasks under the  project file? 
But none of them have disappeared, at least. 

Did you save the buffers? 

```
(setq org-roam-capture-templates
        '(("p" "project")
          ("pv" "version" plain "%?"
           :if-new (file+head "${slug}.org" "#+title: ${title}")
           :unnarrowed t)
          ("pt" "project task" entry "** TODO %?"
           :if-new (file+olp "${slug}.org" ("Tasks")))))
```

![Screenshot from 2021-11-11 11-01-58](https://user-images.githubusercontent.com/12507865/141278604-814ada87-d79b-4e06-a2e8-e3e369ddef31.png)

**Comment by dirodriguezm** [comment]
Hi, thanks so much for helping. Yes I did save my buffers. I can provide more info.

What I want to do is to be able to create tasks in project versions and reference versions in main project file. So yeah, the workflow would require at least 2 files and 3 templates (project, project version and task).

I'm posting the content of my templates 

1. Project template
   ```
   #+title: ${title}
   #+filetags: :Project:

   * General Description
   This section gives an overview of the project, its purpose, main objectives and overall design information that isn't strictly related to its implementation.
   
   %?
   * Versions
   Each product *version* is a project itself. It will have its notes (requirements, design and important remarks), tasks and changelog
   ```

2. Version template
   ```
   #+title: ${title}
   #+filetags: :Project:Version:

   - Project: %?

   * Notes
   * Tasks
   * Changelog
   ```

The thing is that the templates work ok, but after capturing with the task inside a version file, the file disappears from org roam (it disappears from the org roam buffer as well, if I had a back-link)

**Comment by dirodriguezm** [comment]
Further investigation led to the bug being related to whether the "Tasks" heading is the last heading of the file. In the examples of my previous comment, if I remove the "Changelog" heading, then it all works.

**Comment by nobiot** [comment]
Well, I guess. I still can't reproduce the problem.
I used the templates you provided.

```
        ("pt" "project task" entry "** TODO %?"
         :if-new (file+olp "${slug}.org" ("Tasks"))
```
When you have this template for "project task", how do you add tasks to an existing project version like in your screen? I can only create a new files named "Tasks" every time I create a new task...

![image](https://user-images.githubusercontent.com/12507865/141373003-a0b6a513-bc2a-4602-b906-16327ea7f4e7.png)

You don't need to teach me this, or pursue any longer. If you can verify it is a reproducible problem issue, then it should be reported as such.  My inability to reproduce what you do is not relevant.

**Comment by dirodriguezm** [comment]
I'm not sure if you are asking me for more information. I think the real, reproducible issue here is a bit simpler to what I originally posted.

1. Create node using `org-roam-node-find` with this template:
    ```
    ("t" "test" plain "%?"
             :if-new (file+head "projects/${slug}.org" "#+title: ${title}\n* First\n* Second")
             :unnarrowed t)
    ```
    So you end up with a file like this
    ```
    #+title: test

    * First
    * Second
    ```
2. Capture into previous node using `org-roam-capture` and this template:
```
("n" "new" entry "** TODO %?" 
        :if-new (file+olp "${slug}.org" ("First")))
```
So you end up with a file like this
```
#+title: test

* First
* * TODO Captured note
* Second
```

At this point, the file "test" is no longer visible in org-roam, but if we remove the "Second" heading from the template and do the whole process again, it works as expected

**Comment by jethrokuan** [comment]
I also cannot reproduce, `test` still appears for me following your steps in https://github.com/org-roam/org-roam/issues/1962#issuecomment-967078223.

Also your entry template should be just `"* TODO %?"`.

**Comment by dirodriguezm** [comment]
Thanks for trying to help out. Maybe there's something wrong with my setup and that's why nobody else can reproduce. 

Regarding the entry template, I thought that it should be `"** TODO %?"` since it would be a second level heading inside the first level heading called "First", but made the change as you suggested and didn't fix it

**Comment by nobiot** [comment]
> Capture into previous node using org-roam-capture and this template:

I can't capture a task into the test file, following the steps in #1962 (comment).
I end up creating a new file for a task with title "First". 

I can't follow the steps properly.

This is what I mean by "you don't have to teach me" -- I guess I don't know well enough about Org-capture in general.

**Comment by dirodriguezm** [comment]
> I guess I don't know well enough about Org-capture in general

Well, neither do I 😅

**Comment by nobiot** [comment]
If you would like me and others to look at it, could you describe exactly how you move from step 1 to step 2? What command do you use for step 1? Do you C-c C-c to finalize the capture? What command do you use to do step 2? Is it done while you are in the newly create buffer? 

I cannot end up in the final state you describe where the task is created within the project file like this:

#+title: test

* First
* * TODO Captured note
* Second

**Comment by dirodriguezm** [comment]
Sure, In step 1 I call `org-roam-node-find`, name the new node _test_ and select my test template (the one that has First and Second headings), then I close the capture with `C-c C-c`. 

After step 1 you should have a node called _test_ and then you can call `org-roam-capture` from anywhere which should open a pop up buffer to select which node you want to capture to. It could be a new node, or in this case the previously created node (_test_).

Now we select the _new_ template and the capture buffer now opens with a TODO entry. To finish we press `C-c C-c` and the file _test_ should have the new entry. 

But now, If you are me, you will find that _test_ node is no longer available in org-roam :(

**Comment by nobiot** [comment]
@dirodriguezm 

Thank you for the steps. I didn't know this feature of `org-roam-capture`.

Unfortunately for you, though, my Emacs does what you would like. The *test* does not disappear.

![image](https://user-images.githubusercontent.com/12507865/141612152-bf548f83-f2ea-4885-a2c8-07650b0d04fb.png)

One minor thing I notice is that you have a space between the two stars for the "TODO Captured note" line; mine does not insert a space.

> `* * TODO Captured note`

Here is the templates I used -- copy and paste from you.

```
(setq org-roam-capture-templates
      '(("t" "test" plain "%?"
         :if-new (file+head "projects/${slug}.org" "#+title: ${title}\n* First\n* Second")
         :unnarrowed t)
        ("n" "new" entry "** TODO %?" 
         :if-new (file+olp "${slug}.org" ("First")))))
```

**Comment by dirodriguezm** [comment]
@nobiot
Thanks again for checking this out. I'm sad to see that it works fine for you, I wonder what could have I done to break it. 

Regarding this part

> `* * TODO Captured note`

It was just a typo here in the comment.

I'm not sure if I should close this now since I'm the only one that has the issue.

**Comment by nobiot** [comment]
Try without doom?

**Comment by dirodriguezm** [comment]
You are right ! 

I just tried this on a clean Emacs install and it works fine. Then tried on clean Doom install and the problem raised again. 

This confirms that it has something to do with doom and not org-roam
